### PR TITLE
Create our own socket file for socket file test

### PR DESCRIPTION
**Pull request by swt2c**
/dev/log doesn't exist in certain types of containers (e.g., Fedora's build system).  Additionally, this enables us to run this test on
platforms other than Linux.

### CODE REVIEW

File: tests/filename/filenametest.cpp [hunk 1]
```diff
@@ -710,8 +712,24 @@ TEST_CASE("wxFileName::Exists", "[filename]")
...
+        memset(&addr, 0, sizeof(struct sockaddr_un));
+        addr.sun_family = AF_UNIX;
+        strncpy(addr.sun_path, sock.c_str(), sizeof(addr.sun_path) - 1);
+        if (!bind(fd, (struct sockaddr *) &addr, sizeof(struct sockaddr_un)))
+        {
+            wxON_BLOCK_EXIT1(wxRemoveFile, sock);
+            CHECK( wxFileName::Exists(sock, wxFILE_EXISTS_SOCKET) );
+        }
+    }
+    wxFileName::Rmdir(socktempdir);
```
Comment (vadz) [inline tests/filename/filenametest.cpp:732]: This should be done in `wxON_BLOCK_EXIT` too, shouldn't it?

Unless we decide to add `wxTempDir` class (which would do it in its dtor).

File: tests/filename/filenametest.cpp [hunk 2]
```diff
@@ -710,8 +712,24 @@ TEST_CASE("wxFileName::Exists", "[filename]")
     // These files are only guaranteed to exist under Linux.
     // No need for wxFILE_EXISTS_NO_FOLLOW here; wxFILE_EXISTS_SYMLINK implies it
     CHECK( wxFileName::Exists("/proc/self", wxFILE_EXISTS_SYMLINK) );
-    CHECK( wxFileName::Exists("/dev/log", wxFILE_EXISTS_SOCKET) );
 #endif // __LINUX__
+    wxString name = dirTemp.GetPath() + "/socktmpdirXXXXXX";
+    wxString socktempdir = wxString::From8BitData(mkdtemp(name.char_str()));
+    wxString sock = socktempdir + "/socket";
+    struct sockaddr_un addr;
+    int fd = socket(AF_UNIX, SOCK_STREAM, 0);
```
Comment (vadz) [inline tests/filename/filenametest.cpp:720]: I wonder if we could use the Unix socket created by `wxTCPServer` instead of redoing all this manually?

**Review by vadz — COMMENTED** [review]
Thanks, we can merge it with minimal changes with just ensuring that `Rmdir()` is called (in whichever way), but if we can make the code here simpler by using the existing wxIPC stuff, it would be nice. But, again, if not, no big deal.

File: tests/filename/filenametest.cpp (hunk 2 above)
Comment (swt2c) [reply tests/filename/filenametest.cpp:720]: Yep.  I swear I grepped to see if there was existing code that did Unix sockets... :P

File: tests/filename/filenametest.cpp (hunk 1 above)
Comment (swt2c) [reply tests/filename/filenametest.cpp]: Yeah a `wxTempDir` class would be really useful.  I was surprised it didn't exist.

Comment (swt2c) [reply tests/filename/filenametest.cpp]: Allright, my OCD convinced me to add a `wxTempDir` class...

Comment (vadz) [reply tests/filename/filenametest.cpp]: Yay for the OCD!

But there doesn't seem to be such class in this PR, have you forgotten to push it here perhaps? Or do you plan to change this later, i.e. should I merge this already?

Thanks!

Comment (swt2c) [reply tests/filename/filenametest.cpp]: I plan to do it later (probably not until Sunday or Monday).  You can merge this if you think it's ready, or wait.

One question though - do you think the new class should be part of the public API, or just an internal class for wx use?

Comment (vadz) [reply tests/filename/filenametest.cpp]: OK, I'll merge this one already, thanks.

And I do believe that `wxTempDir` would be useful to have in the public API. TIA!

**Pull request closed without merging**